src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -87,34 +92,29 @@ of this software and associated documentation files (the "Software"), to deal
      */
     @SuppressFBWarnings(value="DC_DOUBLECHECK")
     public static DatadogClient getInstance(String url, String logIntakeUrl, Secret apiKey){
-        if(enableValidations){
-            if (url == null || url.isEmpty()) {
-                logger.severe("Datadog Target URL is not set properly");
-                return null;
-            }
-            if (apiKey == null || Secret.toString(apiKey).isEmpty()){
-                logger.severe("Datadog API Key is not set properly");
+        // If the configuration has not changed, return the current instance without validation
+        // since we've already validated and/or errored about the data
+
+        DatadogHttpClient newInstance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+        DatadogHttpClient httpInstance = (DatadogHttpClient) instance;
+        if (httpInstance != null && httpInstance.equals(newInstance)) {
+            if (DatadogHttpClient.failedLastValidation) {
                 return null;
             }
-            if (logIntakeUrl == null || logIntakeUrl.isEmpty()){
-                logger.warning("Datadog Log Intake URL is not set properly");
-            }
+            return instance;
         }
-
-        if(instance == null){
-            synchronized (DatadogHttpClient.class) {
-                if(instance == null){
-                    instance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+        synchronized (DatadogHttpClient.class) {
+            if (enableValidations) {
+                DatadogHttpClient.instance = newInstance;
+                try {
+                    newInstance.validateConfiguration(url, logIntakeUrl, apiKey);
```

Comment:
```suggestion
                    newInstance.validateConfiguration();
```
You don't need the args because it's not static anymore